### PR TITLE
feat(dist): support auto reload config & compress audit log

### PR DESCRIPTION
**Pull request by imbajin**
Also enhance:
- add time strategy to avoid old logs overwriting by current logs
- auto create log folder if it does not exist
- auto delete log by some conditions (e.g: lastModify + fileSize + filecount)
- add some comment for auth config

The effect of configs like this: (audit log will compressed by exceeding 1day or 200MB, if we choose hour as basic file pattern, it will compressed by 1hour)

image

based on hour like this:

image

The diff between old & new log format:
image

TODO: after current `log4j2.xml` stable, then change other configs together

### CODE REVIEW

File: hugegraph-dist/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -13,17 +14,27 @@
...
             bufferSize="524288" immediateFlush="false">
             <ThresholdFilter level="TRACE" onMatch="ACCEPT" onMismatch="DENY"/>
             <!-- Use simple format for audit log to speed up -->
             <PatternLayout pattern="%-d{yyyy-MM-dd HH:mm:ss} - %m%n"/>
-            <SizeBasedTriggeringPolicy size="50MB"/>
+            <DefaultRolloverStrategy max="5"/>
+
+            <Policies>
+                <SizeBasedTriggeringPolicy size="500MB"/>
+                <TimeBasedTriggeringPolicy interval="1" modulate="true" />
```
[the remark this answers is not in the document]
Comment (imbajin) [reply hugegraph-dist/src/main/resources/log4j2.xml:36]: add some comments for configs later (and also add some description & img on [top](https://github.com/hugegraph/hugegraph/pull/1492#issue-664736677))

File: hugegraph-dist/src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -13,17 +14,27 @@
...
+        <!-- Separate & compress audit log, buffer size is 512KB -->
         <RollingRandomAccessFile name="audit" fileName="logs/audit-hugegraph-server.log"
-            filePattern="logs/$${date:yyyy-MM}/audit-hugegraph-server-%d{yyyy-MM-dd}-%i.log"
+            filePattern="logs/$${date:yyyy-MM}/audit-hugegraph-server-%d{yyyy-MM-dd}-%i.gz"
             bufferSize="524288" immediateFlush="false">
             <ThresholdFilter level="TRACE" onMatch="ACCEPT" onMismatch="DENY"/>
             <!-- Use simple format for audit log to speed up -->
             <PatternLayout pattern="%-d{yyyy-MM-dd HH:mm:ss} - %m%n"/>
-            <SizeBasedTriggeringPolicy size="50MB"/>
+            <DefaultRolloverStrategy max="5"/>
```
[the remark this answers is not in the document]
Comment (imbajin) [reply hugegraph-dist/src/main/resources/log4j2.xml:32]: now only support compressed one by one

File: hugegraph-dist/src/assembly/static/conf/log4j2.xml [hunk 3]
```diff
@@ -1,29 +1,65 @@
...
             <ThresholdFilter level="TRACE" onMatch="ACCEPT" onMismatch="DENY"/>
             <!-- Use simple format for audit log to speed up -->
             <PatternLayout pattern="%-d{yyyy-MM-dd HH:mm:ss} - %m%n"/>
-            <SizeBasedTriggeringPolicy size="50MB"/>
+            <!-- Trigger after exceeding 1hour or 500MB -->
+            <Policies>
+                <SizeBasedTriggeringPolicy size="500MB"/>
+                <TimeBasedTriggeringPolicy interval="1" modulate="true" />
+            </Policies>
+            <!-- Keep 2 files/h & auto delete after 60days or over 5GB / 500 files -->
```
Comment (javeme) [inline hugegraph-dist/src/assembly/static/conf/log4j2.xml:52]: Keep 2 files each hour
<after 60 days> or <over 5GB or 500 files>

Comment (imbajin) [reply hugegraph-dist/src/assembly/static/conf/log4j2.xml:52]: use `[after 60 days] or [over 5GB or 500 files]` now (`<>` looks like comment `<!-- -->`)

File: hugegraph-dist/src/assembly/static/conf/rest-server.properties [hunk 4]
```diff
@@ -22,15 +22,19 @@ batch.max_write_threads=0
 #auth.admin_token=
 #auth.user_tokens=[]
 
-server.id=server-1
-server.role=master
-
+# auth server configs (comment to disable)
```
Comment (javeme) [inline hugegraph-dist/src/assembly/static/conf/rest-server.properties:25]: cache-invalid also need to start rpc server

File: hugegraph-dist/src/assembly/static/conf/log4j2.xml [hunk 5]
```diff
@@ -1,29 +1,65 @@
...
+    <properties>
+        <property name="LOG_PATH">logs</property>
+        <property name="FILE_NAME">hugegraph-server</property>
+    </properties>
+
     <appenders>
         <Console name="console" target="SYSTEM_OUT">
             <ThresholdFilter level="DEBUG" onMatch="ACCEPT" onMismatch="DENY"/>
-            <PatternLayout pattern="%-d{yyyy-MM-dd HH:mm:ss} %-5r [%t] [%-5p] %c %x - %m%n"/>
+            <PatternLayout pattern="%-d{yyyy-MM-dd HH:mm:ss} [%t] [%p] %c{1.} - %m%n"/>
```
Comment (imbajin) [inline hugegraph-dist/src/assembly/static/conf/log4j2.xml:13]: current log format like this:
image

removed items:
- cpu time of current thread
- fixed 5 length (useless)
- useless %x (almost)
- full class format --> shortly format

File: hugegraph-dist/src/assembly/static/conf/rest-server.properties [hunk 6]
```diff
@@ -10,21 +10,33 @@ graphs=[hugegraph:conf/hugegraph.properties]
...
-#auth.user_tokens=[]
+
+# StandardAuthenticator mode
+#auth.graph_store=hugegraph
+# auth client config
 #auth.remote_url=127.0.0.1:8899,127.0.0.1:8898,127.0.0.1:8897
 
-server.id=server-1
-server.role=master
+# ConfigAuthenticator mode
```
Comment (javeme) [inline hugegraph-dist/src/assembly/static/conf/rest-server.properties:22]: just for ConfigAuthenticator mode

File: hugegraph-dist/src/assembly/static/conf/rest-server.properties [hunk 7]
```diff
@@ -10,21 +10,33 @@ graphs=[hugegraph:conf/hugegraph.properties]
...
+#auth.admin_token=
+#auth.user_tokens=[]
 
+# rpc group configs
```
Comment (javeme) [inline hugegraph-dist/src/assembly/static/conf/rest-server.properties:26]: rpc group configs of multi graph servers

File: hugegraph-dist/src/assembly/static/conf/rest-server.properties [hunk 8]
```diff
@@ -10,21 +10,33 @@ graphs=[hugegraph:conf/hugegraph.properties]
...
 batch.max_write_threads=0
 
-# authentication
+# authentication configs
+# choose 'com.baidu.hugegraph.auth.StandardAuthenticator' or 'com.baidu.hugegraph.auth.ConfigAuthenticator'
 #auth.authenticator=
-#auth.admin_token=
-#auth.user_tokens=[]
+
+# StandardAuthenticator mode
```
Comment (javeme) [inline hugegraph-dist/src/assembly/static/conf/rest-server.properties:17]: for StandardAuthenticator mode

File: hugegraph-dist/src/assembly/static/conf/rest-server.properties [hunk 9]
```diff
@@ -10,21 +10,33 @@ graphs=[hugegraph:conf/hugegraph.properties]
...
+#auth.admin_token=
+#auth.user_tokens=[]
 
+# rpc group configs
+# rpc server configs
 rpc.server_host=127.0.0.1
 rpc.server_port=8090
 #rpc.server_timeout=30
+
+# rpc client configs (enable to keep cache consistency of multi servers)
```
Comment (javeme) [inline hugegraph-dist/src/assembly/static/conf/rest-server.properties:32]: like enable to keep cache consistency

File: hugegraph-dist/src/assembly/static/conf/log4j2.xml (hunk 5 above)
Comment (Linary) [reply hugegraph-dist/src/assembly/static/conf/log4j2.xml:13]: LGTM

**Review by Linary — APPROVED** [review]

**Review by javeme — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Pull request merged**